src/pages/api/add_event.ts

```diff
@@ -0,0 +1,41 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import prisma from "../../lib/prisma";
+import { Event } from "@prisma/client";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === "POST") {
+    try {
+      const { name, createdBy, location, eventType, status } = req.body;
+      const description = "description abc";
+      const endDate = "2023-12-31T23:59:59.999Z";
+      const startDate = "2023-10-31T23:59:59.999Z";
+
+      const newEvent = await prisma.event.create({
+        data: {
+          name,
+          createdBy,
+          location,
+          startDate,
+          event: {
+            connect: {
+              name: eventType
+            }
+          },
+          status,
+          description,
+          endDate
+        }
+      });
```

Comment:
Could possibly move this out into a 'service class' to abstract the database query out. Have a look at `https://github.com/codersforcauses/repair-labs/blob/main/src/services/repair-request.ts` for an example